src/Sarif.Viewer.VisualStudio/CodeAnalysisResultManager.cs

```diff
@@ -557,5 +558,44 @@ private void RemoveTemporaryFiles()
             // Unit tests will only create one cache.
             return this.RunIndexToRunDataCache.Values.First().RemappedPathPrefixes.ToArray();
         }
+
+        // Extract selected results from original SarifLog.
+        internal SarifLog GetPartitionedLog(IEnumerable<SarifErrorListItem> listItems)
+        {
+            if (listItems.Any())
+            {
+                int runIndex = listItems.First().RunIndex;
+                string guid = Guid.NewGuid().ToString();
+                foreach (SarifErrorListItem item in listItems)
+                {
+                    if (item.SarifResult != null)
+                    {
+                        item.SarifResult.Guid = guid;
```

Comment:
>item.SarifResult.Guid = guid; [](start = 24, length = 29)

create a bool near string guid, set to true here and use below instead of checking for sarifResult

Reply:
refactored